pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -713,7 +713,22 @@ void PdBlock_iterBegin(JSOBJ _obj, JSONTypeContext *tc) {
         return;
     }
 
-    arrays = get_sub_attr(obj, "_mgr", "column_arrays");
+    NPY_DATETIMEUNIT dunit = (PyObjectEncoder *)tc->datetimeUnit;
+    PyObject *date_unit;
+    if (dunit == NPY_FR_s) {
+        date_unit = "s";
+    } else if (dunit == NPY_FR_ms) {
+        date_unit = "ms";
+    } else if (dunit == NPY_FR_us) {
+        date_unit = "us";
+    } else if (dunit == NPY_FR_ns) {
+        date_unit = "ns";
+    }
+
+    PyObject *mgr = PyObject_GetAttrString(obj, "_mgr");
+    PyObject *name = "column_arrays";
+    arrays = PyObject_CallMethodOneArg(mgr, name, date_unit);
+
     if (!arrays) {
```

Comment:
After you are done with `mgr` you will want to `Py_DECREF(mgr)` to avoid memory leaks